src/lib/optimizer/strategy/subquery_to_join_rule.cpp

```diff
@@ -199,6 +199,26 @@ std::pair<SubqueryToJoinRule::PredicatePullUpResult, bool> pull_up_correlated_pr
   return {result, false};
 }
 
+void push_arithmetic_expression_into_subquery(std::shared_ptr<BinaryPredicateExpression>& predicate_expression,
+                                              const std::shared_ptr<ArithmeticExpression>& arithmetic_expression,
+                                              std::shared_ptr<LQPSubqueryExpression>& subquery_expression,
+                                              size_t arithmetic_expression_position, bool subquery_expression_is_left) {
+  // Create a copy of the arithmetic expression which has the subquery expression replaced with the subquery's result.
+  const auto new_arithmetic_expression = std::make_shared<ArithmeticExpression>(
+      arithmetic_expression->arithmetic_operator,
+      subquery_expression_is_left ? subquery_expression->lqp->node_expressions[0]
```

Comment:
Even though a broken LQP should never make it this far, whenever you explicitly take the first element and ignore all others, you should (debug)assert that the vector's size is exactly 1.